### PR TITLE
Feature/assigned tests start assigned test handling

### PR DESCRIPTION
**Pull request by tsiananel**

### CODE REVIEW

File: src/app/services/test-http.service.ts [hunk 1]
```diff
@@ -25,4 +32,8 @@ export class TestHttpService {
   createTest(level: Level): Observable<TestContent> {
     return this.http.post<TestContent>(`${BASE_API_URL}/Test`, { level });
   }
+
+  createAssignedTest(level: Level, testId: string): Observable<TestContent> {
```
Comment (Inunor) [inline src/app/services/test-http.service.ts:36]: is it the correct naming? I mean startAssinedTest would be better as I understand, even API has /start word

File: src/app/services/store/test-store.service.ts [hunk 2]
```diff
@@ -10,22 +10,34 @@ import { Level } from '../../constants/data-constants';
...
 
   testSubject$ = new BehaviorSubject<TestContent | null>(null);
 
-  testResults$ = this.resultsSubject$.asObservable();
+  allTests$ = this.allTestsSubject$.asObservable();
 
   test$ = this.testSubject$.asObservable();
 
+  results$: Observable<TestResult[] | undefined> = this.allTests$.pipe(
+    map((arr) => arr?.filter((i) => i.testPassingDate || null)),
```
Comment (Inunor) [inline src/app/services/store/test-store.service.ts:22]: Please refactor this method, at least give meaningful name to variables. What is arr, i, why null and etc

File: src/app/services/store/test-store.service.ts [hunk 3]
```diff
@@ -10,22 +10,34 @@ import { Level } from '../../constants/data-constants';
...
+  );
+
+  assignedTests$: Observable<TestResult[] | undefined> = this.allTests$.pipe(
+    map((arr) =>
```
Comment (Inunor) [inline src/app/services/store/test-store.service.ts:26]: Same here, it is very difficult to understand what is going on

File: src/app/models/writing-test/writing-test.component.ts [hunk 4]
```diff
@@ -35,4 +35,8 @@ export class WritingTestComponent implements OnInit {
   }
 
   nextTest() {}
+
+  onSpeakingSubmit(): void {
+    // this.speachRecorded.emit(this.topic);
```
Comment (Inunor) [inline src/app/models/writing-test/writing-test.component.ts:40]: ?

Comment (tanyushanel) [reply src/app/models/writing-test/writing-test.component.ts:40]: From another branch)

File: src/app/components/user-results-table/user-results-table.component.html [hunk 5]
```diff
@@ -59,7 +67,7 @@ <h3 class="table-caption">Your recent results</h3>
               <span *ngIf="element.essayMark; else noResult">{{ element.essayMark }}</span>
               <span *ngIf="element.speakingMark; else noResult">{{ element.speakingMark }}</span>
               <ng-template #noResult>
-                <span class="material-icons"> hourglass_empty </span>
+                <span class="material-icons" matTooltip="Pending..."> query_builder </span>
```
Comment (RomPavel) [inline src/app/components/user-results-table/user-results-table.component.html:70]: I would change **Pending...** to **In Review**

File: src/app/models/speaking-test/speaking-test.component.ts [hunk 6]
```diff
@@ -16,6 +24,10 @@ declare let MediaRecorder: any;
 export class SpeakingTestComponent implements OnInit, OnDestroy {
   @Input() speaking: TopicModule | null = null;
 
+  @Input() topic: TopicModule | null = null;
```
Comment (RomPavel) [inline src/app/models/speaking-test/speaking-test.component.ts:27]: what is the diff between Input speaking and Input toping?

File: src/app/models/user-profile/user-profile.component.html [hunk 7]
```diff
@@ -21,11 +22,27 @@ <h3 class="table-caption">Please choose your level to Start Test:</h3>
...
+    <h3 class="table-caption warn">You have a test assigned!</h3>
+    <p class="table-caption">
+      Deadline: <span class="warn">{{ deadLine | date: 'medium' }}</span>
+    </p>
+    <button
+      [disabled]="!selectedLevel || isStarted"
+      mat-raised-button
+      color="primary"
+      class="start-btn"
+      (click)="onStartAssignedButtonClick(selectedLevel, assignedTests$)"
```
Comment (RomPavel) [inline src/app/models/user-profile/user-profile.component.html:35]: I dont like that you pass Observable here

**Review by RomPavel — APPROVED** [review]

**Review by Inunor — APPROVED** [review]

**Pull request merged**